centos5/Dockerfile

```diff
@@ -14,16 +19,300 @@ RUN localedef -i en_US -f UTF-8 en_US.UTF-8 \
 
 USER linuxbrew
 WORKDIR /home/linuxbrew
-
-ENV PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH \
-	SHELL=/bin/bash \
+ENV SHELL=/bin/bash \
 	USER=linuxbrew
 
-# Install Linuxbrew.
-# Create symlinks for GCC 4.4.
-# Install a recent version of file that supports --print0.
-RUN sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)" \
-	&& ln -s /usr/bin/gcc44 ~/.linuxbrew/bin/gcc-4.4 \
-	&& ln -s /usr/bin/g++44 ~/.linuxbrew/bin/g++-4.4 \
-	&& HOMEBREW_BUILD_FROM_SOURCE=1 HOMEBREW_NO_AUTO_UPDATE=1 brew install file-formula \
-	&& brew config
+RUN wget --no-check-certificate -O zlib.tar.gz https://zlib.net/zlib-1.2.11.tar.gz \
+    && mkdir zlib-src \
+    && tar -xf zlib.tar.gz -C zlib-src --strip-components=1 \
+    && cd zlib-src \
+    && ./configure --prefix=/home/linuxbrew/bootstrap/zlib \
+    && make install
```

Comment:
Can we use the host's `zlib` rather than compiling it from source?

Reply:
Yes and no. Here is why. Dockerfiles in our case have two objectives:

1. To be used "as is" for building the images and running the tool (~Linuxbrew~ Homebrew).
2. To be used as installation instructions for that specific platform.

Goal 1 permits usage of host tools. Goal two - does not. The reason I chose to bootstrap `zlib` is simple: I don't think goal 1 is the driver in our case. There is a chance, that all of the stuff that I'm doing here should be a part of a "bootstrapping" Bash script because instructions are identical for all Linux distributions.

Reply:
I feel the purpose of the CentOS 5 `Dockerfile` is to build a working Docker image for CentOS 5 in as simple and concise a method as possible, i.e. goal 2. A general purpose works-on-any distribution bootstrapping shells script is useful and could be kept elsewhere.